rust-examples/cmp.rs

```diff
@@ -0,0 +1,382 @@
+#![feature(lang_items, no_core, fundamental, intrinsics)]
+#![no_core]
+
+fn main() {
+    let x: i32 = 0;
+    let y: i32 = 1;
+
+    // (i32.const 1)
+    wasm::print_i32((x != y) as i32);
+    wasm::print_i32((x.ne(&y)) as i32);
+
+    // (i32.const 0)
+    wasm::print_i32((x == y) as i32);
+    wasm::print_i32((x.eq(&y)) as i32);
+
+    // (i32.const 1)
+    wasm::print_i32((x < y) as i32);
+    wasm::print_i32((x.lt(&y)) as i32);
+
+    // (i32.const 1)
+    wasm::print_i32((x <= y) as i32);
+    wasm::print_i32((x.le(&y)) as i32);
+
+    // (i32.const 0)
+    wasm::print_i32((x > y) as i32);
+    wasm::print_i32((x.gt(&y)) as i32);
+
+    // (i32.const 0)
+    wasm::print_i32((x >= y) as i32);
+    wasm::print_i32((x.ge(&y)) as i32);
+}
+
+pub mod marker {
```

Comment:
as mentioned in the PR it's a slightly simplified cmp.rs from std/core :)
I just removed the derives (which come with their own requirements, eg assert_receiver_is_clone etc, which we may not end up wanting to port, but mostly because the generated impl for some of those traits required u64s and such :) and also I commented most of the cmp.rs impl macros for most if not all of the types other than i32. Quite a modification, but apart from that it's the regular rust traits.
The iterator.rs example however has a reimplementation of IntoInter and Iterator (since the iter support pulls a lot of other stuff which don't work yet)